src/Service/DownloadService.php

```diff
@@ -97,6 +102,41 @@ public function render(array $data): string
 
     }//end render()
 
+    /**
+     * Downloads a docx.
+     * The html that is added has to be whitout a <head><style></style></head> section.
+     *
+     * @param array $data The data to render for this docx.
+     *
+     * @return string The docx as file output.
+     */
+    public function downloadDocx(array $data): string
+    {
+        $raw = $this->render($data);
+
+        $docx    = new PhpWord();
+        $section = $docx->addSection();
+        Html::addHtml($section, $raw);
+        $file = 'data.docx';
+
+        header("Content-Description: File Transfer");
+        header('Content-Disposition: attachment; filename="'.$file.'"');
+        header('Content-Type: application/vnd.openxmlformats-officedocument.wordprocessingml.document');
+        header('Content-Transfer-Encoding: binary');
+        header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+        header('Expires: 0');
+
+        $docxWriter = IOFactory::createWriter($docx, 'Word2007');
+        $fileId     = Uuid::uuid4();
+        $filename   = '/var/tmp/'.$fileId->toString().'.docx';
```

Comment:
Are you sure about the / prefix for var? (Symfony has two var folders, hence the question)

Reply:
if I remove the / prefix for var I get this error message 
<img width="1076" alt="image" src="https://github.com/CommonGateway/CoreBundle/assets/55533449/7ae1414c-549c-4c39-a523-830c1caae23b">

Reply:
Well, in that case I reckon the / is correct :)